encoders/nlp/TransformerTorchEncoder/__init__.py

```diff
@@ -120,9 +139,25 @@ def amp_accelerate(self):
         else:
             return nullcontext()
 
-    def _compute_embedding_from_model_output(self, outputs: 'BaseModelOutput', input_tokens: Dict):
+    def _api_call(self, query):
+        retries = 0
+
+        while retries < self.max_retries:
+            retries += 1
+            headers = {"Authorization": "Bearer " + self.api_token}
+            api_url = f"https://api-inference.huggingface.co/pipeline/feature-extraction/{self.pretrained_model_name_or_path}"
```

Comment:
this `api_url` can be set in the `constructor`. Not need to instatiate at every call N times

Reply:
so does the header

Reply:
I think it is nicer inside this function, but outside the `while` loop. Almost no additional computation, but the API call code is put into just one place.